.github/workflows/build-executable.yml

```diff
@@ -0,0 +1,90 @@
+name: Build Flow-PHP executable
+
+on:
+  pull_request:
+  push:
+    branches: [ 1.x ]
+
+jobs:
+  build-phar:
+    name: "Build PHAR"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        dependencies:
+          - "locked"
+        php-version:
+          - "8.1"
+        operating-system:
+          - "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v3"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: none
+          tools: composer:v2
+          php-version: "${{ matrix.php-version }}"
+          ini-values: memory_limit=-1
+
+      - name: "Get Composer Cache Directory"
+        id: composer-cache
+        run: |
+          echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: "Cache Composer dependencies"
+        uses: "actions/cache@v3"
+        with:
+          path: "${{ steps.composer-cache.outputs.dir }}"
+          key: "php-${{ matrix.php-version }}-locked-composer-${{ hashFiles('**/composer.lock') }}"
+          restore-keys: |
+            php-${{ matrix.php-version }}-locked-composer-
+
+      - name: "Install locked dependencies"
+        run: "composer install --no-interaction --no-progress"
+
+      - name: "Build PHAR file"
+        run: "composer build:phar"
+
+      - name: Validate Flow PHAR
+        run: |
+          ./flow-php.phar --version
+
+      - name: Download SPC bin artifact
+        id: download-spc-artifact
+        uses: "dawidd6/action-download-artifact@v2"
+        with:
+          github_token: ${{secrets.GITHUB_TOKEN}}
+          repo: "crazywhalecc/static-php-cli"
+          branch: main
+          workflow: "release-build.yml"
+          name: "spc-linux-x86_64"
+
+      - name: Validate SPC bin
+        run: |
+          chmod +x spc
+          ./spc --version
+
+      - name: "Download sources"
+        run: ./spc download --with-php=${{ matrix.php-version }} php-src,micro,pkg-config,onig,zlib,openssl,gmp,libiconv,libxml2
+
+      - name: "Build phpmicro Executable"
+        run: |
+          SPC_USE_SUDO=yes ./spc doctor --auto-fix
+          ./spc build pcntl,posix,mbstring,tokenizer,dom,xmlreader,filter,openssl,bcmath,gmp,fileinfo,phar --build-micro
```

Comment:
When this PR (https://github.com/crazywhalecc/static-php-cli/pull/188) lands, we can greatly reduce the time of the build.

Reply:
any chance to put snappy extension here?

Reply:
Error for Snappy + Ubuntu: https://github.com/crazywhalecc/static-php-cli/issues/195